### DISCUSSION TITLE
Is there any refrence implementation available for Linux driver of Athena Crypto Module?

### DISCUSSION THREAD

**Discussion by govindsi** [polarfire-soc/polarfire-soc-documentation#345] [Linux drivers for PolarFire Soc]
I am looking for Athena linux Crypto driver , is there some reference implementation or WIP work available for reference with ICICLE board?

**Comment by vfalanis** [comment]
Hi @govindsi ,

It is worth mentioning that the user crypto is only supported on PolarFire SoC "S" devices (e.g., MPFS250TS). If you are interested in crypto, it might be worth having a look at the PolarFire SoC Video Kit.

To answer your question, we currently don't have a linux user crypto driver published. However this is something we have in our plans. I would suggest to watch/follow our repositories to get notified of new releases.

Thanks

**Comment by govindsi** [comment]
Thank you for the updates, @vfalanis. I am currently working with the MPFS250TS-1FCSG536I SoC. Could you please guide me to the specific reference regarding hardware crypto in the PolarFire SoC Video Kit documentation? Additionally, I'm interested to know if there are any tentative release timelines or schedules for future release with this support.

**Reply by vfalanis** [reply]
Hi @govindsi ,

The user crypto documentation is not publicly available. I would suggest to open a tech support case with regards this request or any other information regarding schedules and timelines.

Thanks

**Reply by sanjeevb-bansal** [reply]
Hi,

Just to add, **PolarFire Family Security User Guide** is available here which is applicable to your device as well:- https://www.microchip.com/content/dam/mchp/documents/FPGA/ProductDocuments/UserGuides/Microchip_PolarFire_FPGA_and_PolarFire_SoC_FPGA_Security_User_Guide_VA%20(2).pdf

Tech support case can be opened using below link:- 
https://microchip.my.site.com/s/
